.github/workflows/auto_assign_issues.yml

```diff
@@ -0,0 +1,40 @@
+# This workflow has been inspired by https://github.com/MrPowers/quinn/blob/main/.github/workflows/assign-on-comment.yml
+
+name: Auto Assign Issues On Comment
+
+on:
+  issue_comment:
+    types: [created]
+
+permissions:
+  issues: write
+
+jobs:
+  auto-assign:
+    runs-on: ubuntu-latest
+    if: (!github.event.issue.pull_request) && github.event.comment.body == 'take'
+    concurrency:
+      # Only run one a time per user
+      group: ${{ github.actor }}-auto-assign-issue
+    steps:
+      - name: Check if issue can be assigned
+        id: check-assignee
+        run: |
+          # Check if the user is already assigned to the issue
+          RESPONSE=$(curl -H "Authorization: token ${{ secrets.GITHUB_TOKEN }}" -LI https://api.github.com/repos/${{ github.repository }}/issues/${{ github.event.issue.number }}/assignees/${{ github.event.comment.user.login }} -o /dev/null -w '%{http_code}' -s)
```

Comment:
We need the GITHUB_TOKEN of a collaborator here?

Reply:
@vikrantpuppala - seems like it. Please refer to the below.

* [Here's the link](https://github.com/nijanthanvijayakumar/unitycatalog/actions/runs/9865325031/job/27242044411) to the action from yesterday, that succeeded when the workflow had the GITHUB_TOKEN of the collaborator.
* [Here's the link to the action](https://github.com/nijanthanvijayakumar/unitycatalog/actions/runs/9887547222/job/27309491522#step:3:17) from today that shows it succeeded but ended up indicating that it `'Requires Authentication'`.

I suspect it didn't work because, _assigning issues is a write operation that modifies repository's data and GitHub requires such requests to be authenticated_. [Reference doco.](https://docs.github.com/en/rest/issues/assignees?apiVersion=2022-11-28#add-assignees-to-an-issue)

Reply:
Sorry, I am a bit confused here, does this mean that only when collaborators comment "take" this would work?

Reply:
Please ignore, just need to ensure that the secret `GITHUB_TOKEN` is setup in the main fork of repo.